packages/cli/src/ceramic-daemon.ts

```diff
@@ -307,154 +189,105 @@ class CeramicDaemon {
    * current behavior, publishing to IPFS. With that change it will make sense
    * to rename this, e.g. `loadDocFromGenesis`
    */
-  async createReadOnlyDocFromGenesis (req: Request, res: Response, next: NextFunction): Promise<void> {
+  async createReadOnlyDocFromGenesis (req: Request, res: Response): Promise<void> {
     const { doctype, genesis, docOpts } = req.body
     const readOnlyDocOpts = { ...docOpts, anchor: false, publish: false }
-    try {
-      const doc = await this.ceramic.createDocumentFromGenesis(doctype, DoctypeUtils.deserializeCommit(genesis), readOnlyDocOpts)
-      res.json({ docId: doc.id.toString(), state: DoctypeUtils.serializeState(doc.state) })
-    } catch (e) {
-      return next(e)
-    }
-    next()
+    const doc = await this.ceramic.createDocumentFromGenesis(doctype, DoctypeUtils.deserializeCommit(genesis), readOnlyDocOpts)
+    res.json({ docId: doc.id.toString(), state: DoctypeUtils.serializeState(doc.state) })
   }
 
   /**
    * Get document state
    */
-  async state (req: Request, res: Response, next: NextFunction): Promise<void> {
-    try {
-      const doc = await this.ceramic.loadDocument(req.params.docid)
-
-      res.json({ docId: doc.id.toString(), state: DoctypeUtils.serializeState(doc.state) })
-    } catch (e) {
-      return next(e)
-    }
-    next()
+  async state (req: Request, res: Response): Promise<void> {
+    const doc = await this.ceramic.loadDocument(req.params.docid)
+    res.json({ docId: doc.id.toString(), state: DoctypeUtils.serializeState(doc.state) })
   }
 
   /**
    * Get all document commits
    */
-  async commits (req: Request, res: Response, next: NextFunction): Promise<void> {
+  async commits (req: Request, res: Response): Promise<void> {
     const docId = DocID.fromString(req.params.docid)
-    try {
-      const commits = await this.ceramic.loadDocumentCommits(docId)
-      const serializedCommits = commits.map((r: any) => {
-        return {
-          cid: r.cid,
-          value: DoctypeUtils.serializeCommit(r.value)
-        }
-      })
-
-      res.json({ docId: docId.toString(), commits: serializedCommits })
-    } catch (e) {
-      return next(e)
-    }
-    next()
+    const commits = await this.ceramic.loadDocumentCommits(docId)
+    const serializedCommits = commits.map((r: any) => {
+      return {
+        cid: r.cid,
+        value: DoctypeUtils.serializeCommit(r.value)
+      }
+    })
+
+    res.json({ docId: docId.toString(), commits: serializedCommits })
   }
 
   /**
    * Apply one commit to the existing document
    */
-  async applyCommit (req: Request, res: Response, next: NextFunction): Promise<void> {
+  async applyCommit (req: Request, res: Response): Promise<void> {
     const { docId, commit, docOpts } = req.body
     if (!(docId && commit)) {
-      res.json({ error: 'docId and commit are required in order to apply commit' })
-      next()
-      return
+      throw new Error('docId and commit are required in order to apply commit')
```

Comment:
Here we can replace the throw with below?

`res.status(400).send('docId and commit are required in order to apply commit')`

Reply:
Or more accurately we should use the same json response format in the error handler middleware to be consistent

Reply:
Let's punt on this change too and just put it in project.